src/content/configuration/optimization.md

```diff
@@ -324,4 +324,37 @@ module.exports = {
     concatenateModules: true
   }
 };
-```
\ No newline at end of file
+```
+
+## `optimization.sideEffects`
+
+`bool`
+
+Tells webpack to recognise the [sideEffects](https://github.com/webpack/webpack/blob/75c6416cfe95217b5ce60d02ba5d24d88c50f538/examples/side-effects/README.md) flag in `package.json` or rules to skip over modules which are flagged to contain no side effects when exports are not used. 
+
+__package.json__
+
+``` json
+{
+  "name": "awesome npm module",
+  "version": "1.0.0",
+  "sideEffects": false
+}
+```
+
+T> Please note that `sideEffects` should be in the big module's `package.json` file and doesn't mean that you need to set `sideEffects` to `false` in your own project's `package.json` which requires that big module.
+
+`optimization.sideEffects` depends on [`optimization.providedExports`](#optimization-providedexports) and [`optimization.usedExports`](#optimization-usedexports). These dependencies have a build time cost, but eliminating modules has positive impact on performance because of less code generation. Effect of this optimization depends on your codebase, try it for possible performance wins.
```

Comment:
I don't think it depends on these optimizations

Reply:
The issue it that static knowledge of `export *` depends on these optimizations.

https://github.com/webpack/webpack/blob/a28f44f613276446fb764dec7fab38b7cff8a07c/lib/optimize/SideEffectsFlagPlugin.js#L72 depends on `safe-reexport`

https://github.com/webpack/webpack/blob/a28f44f613276446fb764dec7fab38b7cff8a07c/lib/dependencies/HarmonyExportImportedSpecifierDependency.js#L157 `export *` is only a `safe-reexport` when at least `providedExports` is enabled.

But you are probably right `usedExports` isn't really needed.